Allow setting a custom layout for ToolTips

eronisko opened the pull request:
This can be done via `toolTip.setLayout(int resource, Context context)`.

The layout is validated for existence of its basic components, so that the currently-existing setters still work.

Please let me know your thoughts on this!

Thanks

eronisko committed:
[BUGFIX] Remove overlay after exit animation using post() to avoid NPEs

eronisko committed:
[FEATURE] Allow setting a custom layout for ToolTips
This can be done via toolTip.setLayout(int resource, Context context).

The layout is validated for its basic components, so that the
currently-existing setters still work.

worker8 commented:
Hi eronisko,

Thanks for the PR. I just tested it, it's working fine~! 

While your implementation is good enough by itself, because it has checking to prevent users from supplying a layout without the required id (such as `toolTip_container`), it also restricts users from having a `fully` customizable layout.

So I thought it would be good if we can let the users to input layout in any type of format instead of having to always use 2 TextViews in a LinearLayout with expected ids. Can you further improve this feature?

You will need to change the code in `setupToolTip()` in `TourGuide.java`. Rough idea is to have an `if..else..` checking, if a custom layout is supplied, we will not look for `R.id.toolTip_container`, `R.id.title`,etc, instead, it will just use the custom layout.

Thanks!

eronisko committed:
Merge remote-tracking branch 'upstream/master'

eronisko committed:
Merge branch 'master' into custom_tooltip_layout

eronisko committed:
[FEATURE] Allow setting a custom layout for ToolTip
This can be done via toolTip.setLayout(int resource).

If a custom layout is set, the currently-existing setters will be ignored.

eronisko commented:
Hi, I have made the changes as you described, sorry for the delay. The Travis build has failed, but I'm guessing that's a separate problem.

eronisko commented:
It looks like this feature was implemented in 1bdb23738d526ec3390dd3aa9b62d3adeda80b44 so I'll go ahead and close this PR.
